lms/services/file.py

```diff
@@ -57,32 +64,30 @@ def upsert(self, file_dicts):
         )
 
     def _file_search_query(
-        self,
-        application_instance,
-        type_,
-        *,
-        lms_id=None,
-        course_id=None,
-        name=None,
-        size=None
+        self, guid, type_, *, lms_id=None, course_id=None, name=None, size=None
     ):
         """Return a `File` query with the passed parameters applied as filters."""
-        query = self._db.query(File).filter_by(
-            application_instance_id=application_instance.id,
-            type=type_,
+        query = (
+            self._db.query(File)
+            .join(ApplicationInstance)
+            .filter(
+                # We don't query by application_instance but any file belonging to an AI with the same GUID
+                ApplicationInstance.tool_consumer_instance_guid == guid,
+                File.type == type_,
+            )
         )
 
         if lms_id:
-            query = query.filter_by(lms_id=lms_id)
+            query = query.filter(File.lms_id == lms_id)
```

Comment:
I reckon once there' a join involved, this style of filter is preferred, it avoid name clashes and it's more explicit.